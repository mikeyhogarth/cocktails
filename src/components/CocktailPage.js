import React, { useEffect } from "react";
import useScrollTop from "../hooks/useScrollTop";
import { connect } from "react-redux";
import { Fade, Box, Grid } from "@material-ui/core";
import { currentCocktailSelector } from "../selectors";
import { withStyles } from "@material-ui/core/styles";
import { bindActionCreators } from "redux";
import { enrichCocktail } from "../actions";
import CocktailDetail from "./CocktailPage/CocktailDetail";
import CocktailVariantList from "./CocktailPage/CocktailVariantList";

const fullHeight = "92vh";

const styles = theme => ({
  cocktailDetail: {
    overflow: "auto",
    [theme.breakpoints.up("sm")]: {
      height: fullHeight
    }
  },
  cocktailDetailContent: {
    padding: theme.spacing(2)
  },
  cocktailImage: {
    ...theme.mixins.toolbar,
    backgroundColor: theme.palette.grey[400],
    backgroundRepeatY: "no-repeat",
    backgroundSize: "cover",
    [theme.breakpoints.up("xs")]: {
      height: fullHeight
    }
  },
  mobileImage: {
    height: "20vh",
    backgroundPosition: "center"
  }
});

const CocktailPage = ({ cocktail, enrichCocktail, classes }) => {
  useScrollTop();
  useEffect(() => {
    enrichCocktail(cocktail);
  }, [enrichCocktail, cocktail]);

  const image = cocktail.enrichment && cocktail.enrichment.image;

  return (
    <>
      <Box
        component="div"
        className={classes.mobileImage}
        display={{ xs: "block", md: "none" }}
        style={{ backgroundImage: `url(${image})` }}
      />
      <Grid container className={classes.root}>
        <Grid className={classes.cocktailDetail} item md={6} xs={12}>
          <div className={classes.cocktailDetailContent}>
            <CocktailDetail cocktail={cocktail} />
            <CocktailVariantList cocktail={cocktail} />
          </div>
        </Grid>
        <Grid item md={6} xs={false}>
          <Fade in={!!image}>
            <div
              style={{
                backgroundImage: `url(${image})`
              }}
              className={classes.cocktailImage}
            />
          </Fade>
        </Grid>
      </Grid>
    </>
  );
};

const mapStateToProps = (state, ownProps) => ({
  cocktail: currentCocktailSelector(state, ownProps)
});

const mapDispatchToProps = dispatch => ({
  enrichCocktail: bindActionCreators(enrichCocktail, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(CocktailPage));
