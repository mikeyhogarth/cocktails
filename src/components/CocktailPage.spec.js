import React from "react";
import renderer from "react-test-renderer";
import { CocktailPage } from "./CocktailPage";
import store from "../store";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { noop } from "lodash";
import cocktails from "../data/cocktails.json";
import { loadDatabase } from "../utilities/db.utils";

beforeAll(async () => {
  await loadDatabase(store);
});

beforeAll(() => {
  // jsdom does not implement scrollTo so we need to mock it.
  window.scrollToMemo = window.scrollTo;
  window.scrollTo = noop;
});

afterAll(() => {
  window.scrollTo = window.scrollToMemo;
  delete window.scrollToMemo;
});

it("does not explode when rendered", async () => {
  await loadDatabase(store);

  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <CocktailPage cocktail={cocktails[0]} enrichCocktail={noop} />
      </MemoryRouter>
    </Provider>
  );

  expect(tree).toMatchSnapshot();
});
