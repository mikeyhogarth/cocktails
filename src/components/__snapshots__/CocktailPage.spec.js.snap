// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not explode when rendered 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-5 makeStyles-mobileImage-4"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />,
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-cocktailDetail-1 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="makeStyles-cocktailDetailContent-2"
      >
        <h1
          className="MuiTypography-root CocktailPage-cocktailTitle-8 MuiTypography-h1"
        >
          Vesper
        </h1>
        <div
          className="CocktailPage-definitions-6"
        >
          <div
            className="MuiChip-root Definition-chip-9 MuiChip-colorSecondary"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              Before Dinner Cocktail
            </span>
          </div>
          <div
            className="MuiChip-root Definition-chip-9 MuiChip-colorSecondary"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              Martini
            </span>
          </div>
          <div
            className="MuiChip-root Definition-chip-9 MuiChip-colorSecondary"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              Lemon twist
            </span>
          </div>
          <div
            className="MuiChip-root Definition-chip-9 MuiChip-colorSecondary"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label"
            >
              Non-vegan
            </span>
          </div>
        </div>
        <div
          className="MuiPaper-root CocktailPage-cocktailDetailPaper-7 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <ul
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            <li>
              <span>
                6 cl
                 
                <strong
                  aria-describedby={null}
                  className="Connect(IngredientDetail)-tooltip-10"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="40% abv"
                >
                  Gin
                </strong>
              </span>
            </li>
            <li>
              <span>
                1.5 cl
                 
                <strong
                  aria-describedby={null}
                  className="Connect(IngredientDetail)-tooltip-10"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="40% abv"
                >
                  Vodka
                </strong>
              </span>
            </li>
            <li>
              <span>
                0.75 cl
                 
                <strong
                  aria-describedby={null}
                  className="Connect(IngredientDetail)-tooltip-10"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="15% abv"
                >
                  Lillet Blonde
                </strong>
              </span>
            </li>
          </ul>
          <br />
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Shake and strain into a chilled cocktail glass.
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
    >
      <div
        className="makeStyles-cocktailImage-3"
        style={
          Object {
            "backgroundImage": "url(undefined)",
            "opacity": 0,
            "visibility": "hidden",
          }
        }
      />
    </div>
  </div>,
]
`;
